### PR TITLE
Update docs

### PR DESCRIPTION
**Pull request by joshbruce**
## Description

This PR helps break up the README into more consumable chunks.

## Submitter

- [ ] Test(s) exist to ensure functionality works (if no new tests added, list which tests cover this functionality); or,
- [x] no tests required for this PR.
  
## Reviewer

- [x] Draft GitHub release notes have been updated.
- [x] case_insensitive_refs is only failing test (remove once CI is in place and all tests pass).
- [x] All lint checks pass (remove once CI is in place).
- [x] Merge PR

### CODE REVIEW

**Comment by Feder1co5oave** [comment]
Josh, this is the third PR you've opened concerning updates to the README. Can you slow down a bit? I cannot follow you around in all these new issues.

Remember if you change your mind on some commits you make in the PR branch you can always rewrite them by `git reset` or `git rebase` and then `git push --force`, and the commits will be re-written. This is generally NOT safe to do on a shared public branch because it can "break" the branching history on the collaborators' local copies, but since this is a WIP and you're the only one pushing, it can be considered safe to do, before merging.

**Comment by joshbruce** [comment]
That's fair. Not sure the other specific PRs you're referring to - I only have two opened at the moment: #1076 (0.3.17 release updating PR templates) and this one.

I believe the others might have been merged or closed...I did run into an issue where GitHub was listing my PRs as being from an "unknown repository," which I've never seen before now; therefore, some of the closings were due to that. Don't think I'll be opening any more new things for a bit. :)

Definitely understand the feeling though - this is what it felt like when @chjj and I were doing the organization thing - things were done before I could hit reply. ;)

File: USAGE_EXTENSIBILITY.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
+## Extending Marked
```
Comment (joshbruce) [inline USAGE_EXTENSIBILITY.md]: @Feder1co5oave, @UziTech, and @styfle: Could use some help on the content here, I think. Purpose is to document and demonstrate overriding/extending the renderer, lexer, and parser...seems we are light on documentation there.

File: USAGE_EXTENSIBILITY.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+To champion the single-reponsibility and open/closed prinicples, we have tried to make it relatively painless to extend marked. If you are looking to add custom functionality, this is the place to start.
+
+<ul>
+	<li><a href="#renderer">The renderer</a></li>
+	<li><a href="#lexer">The lexer</a></li>
+	<li><a href="#parser">The parser</a></li>
+</ul>
+
+<h2 id="rederer">The renderer</h2>
```
Comment (joshbruce) [inline USAGE_EXTENSIBILITY.md]: The renderer...

File: USAGE_EXTENSIBILITY.md [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+- strong(*string* text)
+- em(*string* text)
+- codespan(*string* code)
+- br()
+- del(*string* text)
+- link(*string* href, *string* title, *string* text)
+- image(*string* href, *string* title, *string* text)
+- text(*string* text)
+
+<h2 id="lexer">The lexer</h2>
```
Comment (joshbruce) [inline USAGE_EXTENSIBILITY.md]: The lexer...

File: USAGE_EXTENSIBILITY.md [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+The lexer is...
+
+
+<h2 id="parser">The parser</h2>
```
Comment (joshbruce) [inline USAGE_EXTENSIBILITY.md]: The parser...

File: USAGE_EXTENSIBILITY.md [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+
+The parser is...
+
+***
```
Comment (joshbruce) [inline USAGE_EXTENSIBILITY.md]: The current documentation

**Comment by joshbruce** [comment]
See also #1063

File: USAGE_EXTENSIBILITY.md (hunk 2 above)
Comment (styfle) [reply USAGE_EXTENSIBILITY.md]: Why are you using HTML here? Instead of `<h2>` you can use `## Renderer`

**Comment by joshbruce** [comment]
Another benefit here is that we have to update documentation not in the README, we don't have to publish a new release to NPM (facade pattern for the win??).

File: USAGE_EXTENSIBILITY.md (hunk 2 above)
Comment (joshbruce) [reply USAGE_EXTENSIBILITY.md]: Fair question. For the `id` attribute. Is there another way to get inline anchors to work properly using GitHub Markdown??

Comment (joshbruce) [reply USAGE_EXTENSIBILITY.md]: Consolidating comments from #1061: I've always run into issues there [GitHub automatically generating IDs]; so, normally end up doing it by hand. Also, could make it more difficult to use in a non-GFM setting; at least here we know the TOC will always work[...like pulling in the Markdown file into a separate context outside of the GFM markdown rendering.]

Comment (styfle) [reply USAGE_EXTENSIBILITY.md]: I typically let github generate it like so: https://github.com/styfle/copee#demo
But a TOC is a good point, you don't want to accidentally break it if you change the heading later.

File: AUTHORS.md [hunk 6]
```diff
@@ -15,8 +15,8 @@ To be listed here, just make a contribution and, if it has significant impact, t
 
 Committers are contributors who also have the responsibility, privilege, some might even say burden of being able to review and merge contributions (just usually not they're own).
 
-- Tony Brix @UziTech
-- Federico Soave @Feder1co5oave
+- Tony Brix @UziTech - really good at the test harness and environment setup
+- Federico Soave @Feder1co5oave - really good at the regex things
 - Steven @styfle
```
Comment (styfle) [inline AUTHORS.md]: > Steven @styfle - good for nothing

😜

Comment (joshbruce) [reply AUTHORS.md]: I honestly thought about putting: Good for lying in the weeds and randomly showing with a nugget of excess truth on operations before disappearing again. 😝

**Pull request merged**